aws-xray-recorder-sdk-sql/src/main/java/com/amazonaws/xray/sql/SqlSubsegments.java

```diff
@@ -36,43 +36,55 @@
+
+    public static final String DRIVER_VERSION = "driver_version";
+    public static final String DATABASE_TYPE = "database_type";
+    public static final String DATABASE_VERSION = "database_version";
+    public static final String SANITIZED_QUERY = "sanitized_query";
+    public static final String DEFAULT_DATABASE_NAME = "database";
 
     /**
      * Begins a {@link Subsegment} populated with data provided by the {@link Connection#getMetaData} method. Includes
-     * the SQL query string if it is non-null, omits it otherwise.
+     * the SQL query string if it is non-null, omits it otherwise. Takes care to swallow any potential
+     * {@link SQLException}s and always start a subsegment for consistency.
      *
      * @param connection the JDBC connection object used for the query this {@link Subsegment} represents.
      * @param query the SQL query string used in this query, or {@code null} if it is not desirable to include in the
      *              subsegment, e.g. for security concerns.
      * @return the created {@link Subsegment}.
-     * @throws SQLException if there is a bad interaction with the {@link Connection}.
      */
-    public static Subsegment forQuery(Connection connection, @Nullable String query) throws SQLException {
-        DatabaseMetaData metadata = connection.getMetaData();
+    public static Subsegment forQuery(Connection connection, @Nullable String query) {
+        DatabaseMetaData metadata = null;
         String subsegmentName = DEFAULT_DATABASE_NAME;
+
         try {
+            metadata = connection.getMetaData();
+            String database = connection.getCatalog();
             URI normalizedUri = new URI(new URI(metadata.getURL()).getSchemeSpecificPart());
-            subsegmentName = connection.getCatalog() + "@" + normalizedUri.getHost();
+            subsegmentName = database + "@" + normalizedUri.getHost();
         } catch (URISyntaxException e) {
             logger.debug("Unable to parse database URI. Falling back to default '" + DEFAULT_DATABASE_NAME
-                    + "' for subsegment name.", e);
+                + "' for subsegment name.", e);
+        } catch (SQLException e) {
+            logger.debug("Encountered exception while retrieving metadata for SQL subsegment "
+                + ", starting blank subsegment instead");
+            return AWSXRay.beginSubsegment(subsegmentName);
         }
 
         Subsegment subsegment = AWSXRay.beginSubsegment(subsegmentName);
-
         subsegment.setNamespace(Namespace.REMOTE.toString());
-        subsegment.putSql(URL, metadata.getURL());
-        subsegment.putSql(USER, metadata.getUserName());
-        subsegment.putSql(DRIVER_VERSION, metadata.getDriverVersion());
-        subsegment.putSql(DATABASE_TYPE, metadata.getDatabaseProductName());
-        subsegment.putSql(DATABASE_VERSION, metadata.getDatabaseProductVersion());
+
+        try {
+            subsegment.putSql(URL, metadata.getURL());
```

Comment:
I don't think we need to assert every one of these but one of them to give some code coverage is still a good idea.